### PR TITLE
mpl: Revamp backtrace support

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

Backtrace support configuration was using hand-rolled configuration checks instead of builtin autoconf macros, and implementation was scattered across files making it hard to follow. This PR cleans up both aspects to make it easier to grok.

## Expected Performance Changes

None.

## Known Issues

None.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Passes tests (included warning check)
* [x] Confirm whitespace/style checkers are happy (or has a good reason for being bad)
* [x] Commits are self-contained and do not do two things at once
* [x] Remove xfail from the test suite when fixing a test
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by raffenet** [comment]
@roblatham00 this is a start to the backtrace cleanup I mentioned. Note that I also removed intermediate buffering of the actual trace before sending it to `stderr`. Was there some reason for that previously? I should probably separate it into a separate commit in any case.

**Comment by raffenet** [comment]
Get a sanity check.
test:jenkins/ch3/tcp

**Comment by roblatham00** [comment]
You should check with @gcongiu as he is responsible for the unrolling:  https://github.com/pmodels/mpich/pull/2798  -- not sure what was up with the PGI compiler in 2017, or if whatever required this workaround still exists.

**Comment by raffenet** [comment]
Ah, interesting. I'll follow up with him.

**Comment by gcongiu** [comment]
@raffenet to be honest I don't remember why exactly I fixed things the way I did back then. I think the old logic was falsely detecting `libunwind` only using headers and subsequently causing the missing references at link time. I am not sure why I did the manual unroll of the check instead of using autotools MACROs instead.

**Comment by raffenet** [comment]
After some more thought and wrestling with the implementation, I'm now proposing to remove libbacktrace and libunwind support. Since these have given users headaches and we don't regularly test them, I'd rather just rely on a execinfo, which is easier to verify.

**Comment by raffenet** [comment]
test:jenkins/ch3/tcp

**Comment by raffenet** [comment]
I think the end result here is something easier to support that still covers a good cross section of users who may be interested in a backtrace upon crash. Ready for review.
